metricbeat/helper/server/http/http_test.go

```diff
@@ -22,17 +22,86 @@ package http
 import (
 	"bytes"
 	"context"
+	"crypto/tls"
 	"fmt"
+	"io/ioutil"
+	"net"
 	"net/http"
+	"strconv"
 	"testing"
-	"time"
-
-	"github.com/elastic/beats/metricbeat/helper/server"
 
 	"github.com/stretchr/testify/assert"
+
+	"github.com/elastic/beats/metricbeat/helper/server"
 )
 
-func GetHttpServer(host string, port int) (server.Server, error) {
+var flagtests = []struct {
```

Comment:
awesome,

let's rename, I woul say calling this flag-* would make people think this is a test flag, which it isn't, so something like `cases` would work.

Also, cases should be hosted inside the testing function, not global. Let's move the var inside the `TestHTTPServer` function.

Reply:
Agreed